platform/broadcom/sonic-platform-modules-dell/n3248te/sonic_platform/eeprom.py

```diff
@@ -25,37 +25,40 @@ def __init__(self):
         self.eeprom_path = f
         super(Eeprom, self).__init__(self.eeprom_path, 0, '', True)
         self.eeprom_tlv_dict = dict()
-        try:
-            self.eeprom_data = self.read_eeprom()
-        except Exception:
-            self.eeprom_data = "N/A"
-            raise RuntimeError("Eeprom is not Programmed")
-
-        eeprom = self.eeprom_data
+        
+        if os.geteuid() == 0:
+            try:
+                self.eeprom_data = self.read_eeprom()
+            except Exception:
+                self.eeprom_data = "N/A"
```

Comment:
@justindthomas can we keep the default value `self.eeprom_data = "N/A"` outside of the if condition at the beginning of the of __init__?